src/Swift.Bindings/src/Emitter/StringCSharpEmitter/Handler/TypeHandler.cs

```diff
@@ -348,4 +353,87 @@ public void Emit(IndentedTextWriter writer, IEnvironment env, Conductor conducto
             writer.WriteLine("}");
         }
     }
+
+    class ISwiftObjectMethodWriter
+    {
+        private readonly IndentedTextWriter _writer;
+        private readonly ITypeDatabase _typeDatabase;
+        private readonly ModuleDecl _moduleDecl;
+        private readonly StructDecl _structDecl;
+
+        public ISwiftObjectMethodWriter(IndentedTextWriter writer, ITypeDatabase typeDatabase, ModuleDecl moduleDecl, StructDecl structDecl)
+        {
+            _writer = writer;
+            _typeDatabase = typeDatabase;
+            _moduleDecl = moduleDecl;
+            _structDecl = structDecl;
+        }
+
+        public void WriteNonFrozenStructImplementation()
+        {
+            WriteGetTypeMetadata();
+            WriteNewFromPayloadNonFrozenStruct();
+            WriteMarshalToSwift();
+        }
+
+        public void WriteFrozenStructImplementation()
+        {
+            WriteGetTypeMetadata();
+            WriteNewFromPayloadFrozenStruct();
+            WriteMarshalToSwift();
+        }
+
+        private void WriteGetTypeMetadata()
+        {
+            _writer.WriteLine("public static TypeMetadata GetTypeMetadata() => PInvoke_getMetadata();");
```

Comment:
In `SwiftOptional`, I made a conscious choice to implement the `ISwiftObject` explicitly who avoids polluting the API of each object. Let's do that here too.

Reply:
I am wondering about this one. Making a static interface method explicitly implemented makes in inaccessible within the type implementing the method, right?

However we might need to use `TypeMetadata` within the type. For example to get the size of non-frozen struct at runtime. The following will not work anymore:

```csharp
static TypeMetadata GetTypeMetadata() => ...

static nint PayloadSize => GetTypeMetadata.Size()
```

I guess we could introduce some private methods, is this an approach you'd prefer?

Reply:
To get the type metadata, I used `var metadata = SwiftObjectHelper<SwiftOptional<T>>.GetTypeMetadata();`